### PR TITLE
Add enums to player permutation tests

### PR DESCRIPTION
**Pull request by cce**
## Summary

This made it easier for me to understand the different 7 initial player states * 14 simulated events when reading through each permutation's expected actions. I hope the enum names I picked are accurate — would appreciate any feedback on better names.

## Test Plan

This updates an existing test.

### CODE REVIEW

File: agreement/player_permutation_test.go [hunk 1]
```diff
@@ -86,7 +98,7 @@ func getPlayerPermutation(t *testing.T, n int) (plyr *player, pMachine ioAutomat
 		pM.Children[r].Children[p].ProposalTracker.Staging = pV
 		pM.Children[r].Children[p].ProposalTrackerContract.SawOneVote = true
 		pM.Children[r].Children[p].update(0)
-	case 5: // already reached cert threshold
+	case playerSameRound_ReachedCertThreshold: // already reached cert threshold
 		plyr, pMachine, helper = setupP(t, r, p, soft)
```
Comment (cce) [inline agreement/player_permutation_test.go:102]: Related, should this be `setupP(t, r, p, cert)`?

Comment (nicholasguoalgorand) [reply agreement/player_permutation_test.go:102]: The more common scenario is probably when its in the `cert` step, but it's also possible for this to happen in the `soft` scenario. Feel free to change it to `cert`

File: agreement/player_permutation_test.go [hunk 2]
```diff
@@ -325,409 +356,385 @@ func expectDisconnect(t *testing.T, trace ioTrace, errMsg string, playerN int, e
...
-			}
+			bun := unauthenticatedBundle{Round: r, Period: p, Proposal: pV}
 			ca := cryptoAction{T: verifyBundle, M: message{Bundle: bundle{U: bun, Votes: votes}, UnauthenticatedBundle: bun}, TaskIndex: 0}
-			require.Truef(t, trace.Contains(ev(ca)), "Player should emit action, player: %v, event: %v", playerN, eventN)
+			requireTraceContains(t, trace, ev(ca), playerN, eventN)
 			//ea := ensureAction{Certificate: Certificate(bun), Payload: *payload}
-			//require.Truef(t, trace.Contains(ev(ea)), "Player should emit action, player: %v, event: %v", playerN, eventN)
-		case 10, 11, 12:
-			require.Equalf(t, trace.countAction(), 1, "Plyaer should not emit extra actions, player: %v, event: %v", playerN, eventN)
+			//requireTraceContains(t, trace, ev(ea), playerN, eventN)
```
Comment (nicholasguoalgorand) [inline agreement/player_permutation_test.go:424]: can you do me a favor and remove these commented out lines?

**Review by nicholasguoalgorand — APPROVED** [review]
Looks good.

File: agreement/player_permutation_test.go (hunk 1 above)
Comment (cce) [reply agreement/player_permutation_test.go:102]: OK, AFAICT in the `player.handle()` and `player.handleMessageEvent()` paths exercised by these tests, the only place the `p.Step` value is checked is in a clause `p.Step <= cert` so it doesn't matter.

**Pull request merged**